### ISSUE TITLE
macOS update 10.15.7 broke nix

### ISSUE THREAD

**Issue by Cmdv** [NixOS/nix#4083]
**Describe the bug**

Having recently updated macOS Catalina 10.15.7 I no longer seem to have nix related command available or on the PATH.

```Bash
> echo $PATH
/Users/cmdv/bin:/usr/local/bin:/Users/cmdv/.emacs.d/bin:/usr/bin:/bin:/usr/sbin:/sbin

> echo $NIX_PATH
/Users/cmdv/.nix-defexpr/channels
```
inside my `.zshrc` I still have `. "$HOME/.nix-profile/etc/profile.d/hm-session-vars.sh"` also my mounted volume `/nix` is still present with all it's content.

I tried to run the deamon with `--no-modify-profile` thinking it might 
generate any missing files but that fails and asks me to remove all of nix.

I would remove nix but I have 3 built versions of ghc which each took me around 8 hours to build so I'm not too keen to go down that route 😢 

**Additional context**

Any alternative option than restarting from fresh?

**Comment by Cmdv** [comment]
After lots of trial and error found that was missing is:
```
source /nix/var/nix/profiles/default/etc/profile.d/nix-daemon.sh
```
inside my .zshrc, can't see why updating OS would have edited that file, or is it a case that it's normally put somewhere else?

**Comment by abathur** [comment]
I'm not _certain_ this was your issue, but I think the source statement was actually in /etc/zshrc (so that it would apply for all users, I think), and macOS updates have indeed been observed replacing this file with new copies.  I believe the underlying issue here was fixed in the course of addressing #3456, but that fix hasn't been released yet.

Putting it in ~/.zshrc is _probably_ fine as long as other users don't need to use your system with Nix access.
